packages/babel-parser/src/parser/comments.ts

```diff
@@ -102,7 +102,10 @@ function adjustInnerComments(
 export default class CommentsParser extends BaseParser {
   addComment(comment: Comment): void {
     if (this.filename) comment.loc.filename = this.filename;
-    this.state.comments.push(comment);
+    const { commentsLen } = this.state;
+    if (this.comments.length != commentsLen) this.comments.length = commentsLen;
+    this.comments.push(comment);
+    this.state.commentsLen++;
```

Comment:
Can it happen that we clone the state, push a comment to `.comments`, reset to the previous state, then never call `.addComment` again and so we never do `this.comments.length = commentsLen` removing the new extra comment?